Sources/SignalProducer.swift

```diff
@@ -1830,17 +1833,43 @@ extension SignalProducerProtocol {
 		let lifetime = Lifetime(lifetimeToken)
 
 		let state = Atomic(ReplayState<Value, Error>(upTo: capacity))
+		let start = Atomic(true)
 
-		let start: Atomic<(() -> Void)?> = Atomic {
-			// Start the underlying producer.
+		// Start the underlying producer.
+		func startHandler() {
```

Comment:
`replayLazily` retains `self` only when it is not ever started. Once it is started, it throws the start handler (which retained `self`) away. This is why there is a `start` of type `Atomic<(() -> Void)?>`. The behaviour — at least for the default policy — should not be changed.